src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpNamingStyleDiagnosticAnalyzer.cs

```diff
@@ -1,16 +1,18 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Immutable;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
 
 namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.NamingStyles
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    internal sealed class CSharpNamingStyleDiagnosticAnalyzer : NamingStyleDiagnosticAnalyzerBase
+    internal sealed class CSharpNamingStyleDiagnosticAnalyzer : NamingStyleDiagnosticAnalyzerBase<SyntaxKind>
     {
-        public CSharpNamingStyleDiagnosticAnalyzer()
-        {
-
-        }
+        protected override ImmutableArray<SyntaxKind> SupportedSyntaxKinds { get; } =
+            ImmutableArray.Create(
+                SyntaxKind.VariableDeclarator,
+                SyntaxKind.ForEachStatement,
+                SyntaxKind.SingleVariableDesignation);
```

Comment:
:memo: This is missing several cases. See here:
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/1c3c14d47a36735ebac3eea13cf9d6172fad7157/StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SA1312VariableNamesMustBeginWithLowerCaseLetter.cs#L62-L70

Reply:
```c#
context.RegisterSyntaxNodeAction(VariableDeclarationAction, SyntaxKind.VariableDeclaration);`
```
This is handled `SyntaxKind.VariableDeclarator`.

```c#
context.RegisterSyntaxNodeAction(CatchDeclarationAction, SyntaxKind.CatchDeclaration);
```
You're right, I'll fix this one.
```c#
context.RegisterSyntaxNodeAction(QueryContinuationAction, SyntaxKind.QueryContinuation);
context.RegisterSyntaxNodeAction(FromClauseAction, SyntaxKind.FromClause);
context.RegisterSyntaxNodeAction(LetClauseAction, SyntaxKind.LetClause);
context.RegisterSyntaxNodeAction(JoinClauseAction, SyntaxKind.JoinClause);
context.RegisterSyntaxNodeAction(JoinIntoClauseAction, SyntaxKind.JoinIntoClause);
```
These are not supported as explained in the tests. They're not actual represented as locals in the language. There should later be a design descussion about whether we should pretend that these are in fact locals and do some workarounds to treat `IRangeVariableSymbol`s as `ILocalSymbol`s in the code, or whether we should add yet another option for these. I hope it's not a requirement to sort this out right now.

Reply:
I personally don't see a reason to have a separate naming style for range variables and don't think other people will need this either. But I've heard in the other PR that there has been some discussion about extending symbol specifications and changing them to blur the distinction between 2 kinds of symbols might hinder that. So I'm leaving this for the team to decide.

Reply:
I'm not able to get the variable symbol in VB from a CatchBlock or a CatchStatement.

Reply:
catch clause added in 09d2e08c653a51fc65d6c6d778f6d5edef6e7632